pkg/services/publicdashboards/service/service.go

```diff
@@ -333,6 +333,18 @@ func (pd *PublicDashboardServiceImpl) GetOrgIdByAccessToken(ctx context.Context,
 }
 
 func (pd *PublicDashboardServiceImpl) Delete(ctx context.Context, uid string) error {
+	// get existing public dashboard if exists
+	existingPubdash, err := pd.store.Find(ctx, uid)
+	if err != nil {
+		return ErrInternalServerError.Errorf("Delete: failed to find public dashboard by uid: %s: %w", uid, err)
+	} else if existingPubdash == nil {
+		return ErrPublicDashboardNotFound.Errorf("Delete: public dashboard not found by uid: %s", uid)
+	}
+
+	// validate the public dashboard belongs to the dashboard
+	if existingPubdash.DashboardUid != uid {
```

Comment:
The `uid` identifies the public dashboard so it is different from the `DashboardUid`. In order to perform this validation here you would need to send both values from the `api` layer and then, compare the value obtained from the database (existingPubdash.DashboardUid) with the one provided in the URL.

Reply:
yes, I changed the same